conans/client/manifest_manager.py

```diff
@@ -27,8 +27,8 @@ def check_graph(self, graph, verify, interactive):
 
     def _handle_recipe(self, node, verify, interactive):
         ref = node.ref
-        export = self._cache.export(ref)
-        exports_sources_folder = self._cache.export_sources(ref)
+        export = self._cache.package_layout(ref).export()
+        exports_sources_folder = self._cache.package_layout(ref).export_sources()
```

Comment:
`self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

Reply:
Actually this was a bug! It should be short_paths=None, in order to have the manifest check feature to work properly when there are short_paths.

There were other hidden bug in manifest manager (this is a not very used feature), I have also solved it, adding a test.